src/WebJobs.Script.Abstractions/Description/Binding/BindingMetadata.cs

```diff
@@ -81,6 +93,7 @@ public static BindingMetadata Create(JObject raw)
             string bindingDirectionValue = (string)raw["direction"];
```

Comment:
This is an issue that exists before this PR, but I am actually a bit confused by the goal of this function. It seems to do a mix of manual and automatic deserialization? Am I correct in that there is some redundant assignments? The `raw.ToObject<BindingMetadata>()` along with the `string bindingType = (string)raw["type"];` and similar lines seem redundant? Won't the `ToObject<>()` call already set those properties?

I am curious why we don't just have a custom json converter?

Reply:
My other comment has a suggestion on how to refactor this method. I'll leave it up to you if you want to bother with that or not.

Reply:
Yeah it is confusing, I can give your suggestion a quick try and if there's too much friction we can opt to address this in a different PR

Reply:
@jviau was able to make your refactor suggestion work, will all the tests green - any additional feedback?